src/ast/passes/codegen_llvm.cpp

```diff
@@ -790,6 +790,46 @@ void CodegenLLVM::visit(Call &call)
     expr_ = buf;
     expr_deleter_ = [this, buf]() { b_.CreateLifetimeEnd(buf); };
   }
+  else if (call.func == "pton")
+  {
+    auto af_type = AF_INET;
+    int addr_size = 4;
+    std::string addr = bpftrace_.get_string_literal(call.vargs->at(0));
+    if (addr.find(":") != std::string::npos)
+    {
+      af_type = AF_INET6;
+      addr_size = 16;
+    }
+
+    llvm::Type *array_t = ArrayType::get(b_.getInt8Ty(), addr_size);
+    AllocaInst *buf;
+    if (af_type == AF_INET6)
+    {
+      buf = b_.CreateAllocaBPF(array_t, "addr6");
+    }
+    else
+    {
+      buf = b_.CreateAllocaBPF(array_t, "addr4");
+    }
+
+    char dst[addr_size];
+    Value *octet;
+    auto ret = inet_pton(af_type, addr.c_str(), &dst);
+    if (ret != 1)
+    {
+      LOG(ERROR) << "inet_pton() call returns " << ret;
```

Comment:
I think that it's too late to detect this error here. It currently works like this:
```
# bpftrace -e 'BEGIN { printf("%d\n", pton("1920.168.0.1")[0]); }'
ERROR: inet_pton() call returns 0
Attaching 1 probe...
16
```
which I don't think is a correct behaviour.

If the user enters an invalid address, he should be properly notified what (and where) is wrong without continuing to run the program. I'd suggest to run `inet_pton` in semantic analyser, too (it should be cheap, so no problem running it twice) and detect this failure there.

Reply:
And now this can (and should) be a `FATAL` error.